app/controllers/container_topology_controller.rb

```diff
@@ -22,6 +22,11 @@ def data
     render :json => {:data => generate_topology(params[:id])}
   end
 
+  def refresh_provider
+    render :nothing => true
+    EmsRefresh.queue_refresh([params[:id]])
```

Comment:
@abonas, there are two things that I know of that should be handled before calling `EmsRefresh.refresh(ems)`:

1. @dclarizio pointed out that the UI code handles RBAC constraints to decide whether the refresh button should be shown, and the controller should validate RBAC before issuing the call to the Model, too in order to prevent circumventing the check.
2. The UI currently calls `Model.refresh_ems(array_of_ids)`, which is defined in `VmOrTemplate`, `Host`, and `ExtManagementSystem`.  This call will ensure that the EMS has valid credentials before finally issuing the call to `EmsRefresh.refresh(ems)`.

Hope that helps.